monitoring/metricsreporter/metrics_reporter.go

```diff
@@ -188,6 +200,12 @@ func New(opts ...Option) *MetricsReporter {
 		}
 	}
 
+	// reset  metric every hour because it can grow infinitely
+	async.Interval(ctx, 8*time.Hour, func() {
```

Comment:
Rather than reseting the metrics (which might disrupt the Grafana chart), I think we can just call `DeleteLabelValues` on a `peer_id` after we've disconnected from it. That way, you'd only have 60 (max peers) different values top per metric.

Reply:
Rather than resetting the metrics (which might disrupt the Grafana chart), I think we can just call DeleteLabelValues on a peer_id after we've disconnected from it. That way, you'd only have 60 (max peers) different values top per metric.

Reply:
@nkryuchkov Infinitely growing values (a counter) for a given metric is fine. What we are discussing here as an issue is an infinitely growing amount of labels, e.g.
```
metric{key="peer1"} 0
metric{key="peer2"} 0
...
```

Prometheus doesn't deal well with these case scenarios at large scale.